DirectEmployers/microsite-js
# Reactor build job detail url.

## What does this PR do?
* Refactor `buildJobDetailUrl` to address https://github.com/DirectEmployers/microsite-template/issues/64 

The refactors include just accepting the entire job so that it can build the slugs from existing slug data on the job for both google talent and solr.

## Notes

I have backported these changes to `quality-control` in separate PR https://github.com/DirectEmployers/microsite-js/pull/272 
to address the issue in our main nuxt branches.


To test google talent, checkout/run:

 https://github.com/DirectEmployers/microsites-jobs-psdconnect-org

To test solr, checkout/run:

https://github.com/DirectEmployers/microsites-usnlx-com


Confirm that the job detail links are rendering as before and the Shanghai problem is addressed on usnlx.
